packages/core/src/resolve-namespace-factories.ts

```diff
@@ -10,25 +10,43 @@ export function packageNamespaceFactory(
     }: { dirname(path: string): string; relative(from: string, to: string): string },
     hashSalt = '',
     prefix = '',
-    normalizeVersion = (semver: string) => semver
+    normalizeVersion = (semver: string, _packageName: string) => semver,
+    buildNamespace = defaultBuildNamespace
 ): typeof processNamespace {
     return (namespace: string, stylesheetPath: string) => {
         const configPath = findConfig('package.json', { cwd: dirname(stylesheetPath) });
         if (!configPath) {
             throw new Error(`Could not find package.json for ${stylesheetPath}`);
         }
         const config = loadConfig(configPath) as { name: string; version: string };
-        const fromRoot = relative(dirname(configPath), stylesheetPath).replace(/\\/g, '/');
-        return (
-            prefix +
-            namespace +
-            murmurhash3_32_gc(
-                hashSalt + config.name + '@' + normalizeVersion(config.version) + '/' + fromRoot
-            )
+        return buildNamespace(
+            hashSalt,
+            prefix,
+            namespace,
+            config.name,
+            normalizeVersion(config.version, config.name),
+            relative(dirname(configPath), stylesheetPath).replace(/\\/g, '/')
         );
     };
 }
 
+export function defaultBuildNamespace(
+    hashSalt: string,
+    prefix: string,
+    namespace: string,
+    packageName: string,
+    packageVersion: string,
+    relativeFilePathFromPackageRoot: string
+) {
+    return (
+        prefix +
+        namespace +
+        murmurhash3_32_gc(
```

Comment:
what about `murmurhash3_32_gc`? I'm not sure why I should use it but if it is important it would be good to have a way to reuse the same function you're using

Reply:
Wrote about exporting it or pass it as an argument. You will have access to it in one way or another